src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,400 @@
+---
+title: Writing a Scaffold
+sort: 3
+contributors:
+  - pranshuchittora
+---
+
+Welcome to an demonstration of the new `webpack init` feature! To view what we are building today, simply do `webpack init webpack-scaffold-demo`. This demo is to show you how to build your own webpack scaffold. Let's start by creating an file named `generator.js`.
+
+
+
+
+## Create a basic structure
+
+First, we have to define our skeleton. In order for the CLI to find our options, we've got to define some properties in the constructor first.
+
+__generator.js__
+
+```js
+const Generator = require('yeoman-generator');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {}
+    };
+  }
+};
+```
+
+As you can see, the `configuration` object has to have one property you name, it can be anything. A good practise is to name the underlying property with the name you want to give your `webpack.config.js` file, to better indicate what values each file has.
+
+
+## Make it interactive
+
+In order for us to interact with the users, we make good use of the [`prompting`](http://yeoman.io/authoring/user-interactions.html) method yeoman has. In this method we can get various of answers from the user, like asking for entry points or plugins. You can either manually create each object representing a question, or you can make good use of our utilities from [`webpack-scaffold`](https://github.com/webpack/webpack-cli/tree/master/packages/webpack-scaffold). I'm in a good mood today, so I'm going to build a configuration if the user chooses `Pengwings`.
```

Comment:
1. "We can get various answers" (remove "of")
2. I'm going to build a configuration **only** if the user chooses `Pengwings`